resharper/resharper-unity/src/CSharp/Feature/Services/TypeConversionHints/CSharpTypeConversionHintsSettingsProvider.cs

```diff
@@ -0,0 +1,37 @@
+using JetBrains.Application.InlayHints;
+using JetBrains.Application.Settings;
+using JetBrains.Application.Settings.Implementation;
+using JetBrains.ProjectModel;
+using JetBrains.ReSharper.Feature.Services.CSharp.TypeConversionHints;
+using JetBrains.ReSharper.Plugins.Unity.Settings;
+using JetBrains.Util;
+
+namespace JetBrains.ReSharper.Plugins.Unity.CSharp.Feature.Services.TypeConversionHints
+{
+    [SolutionComponent]
+    public class CSharpTypeConversionHintsSettingsProvider : IUnitySolutionSettingsProvider
+    {
+        private readonly ISettingsSchema mySettingsSchema;
+        private readonly ILogger myLogger;
+
+        public CSharpTypeConversionHintsSettingsProvider(ISettingsSchema settingsSchema, ILogger logger)
+        {
+            mySettingsSchema = settingsSchema;
+            myLogger = logger;
+        }
+
+        public void InitialiseSolutionSettings(ISettingsStorageMountPoint mountPoint)
```

Comment:
I am not sure, that we should change default Rider behaviour for hints, at least in latest EAP

Reply:
This is the change I'm least sure of. I would much rather Rider showed implicit conversion hints by default, but it doesn't - they're hidden. 

My argument for doing it is that implicit type conversion hints are massively useful for highlighting issues such as possible data loss converting `Vector2` to `Vector3` and vice versa. These currently happen silently, and could easily be bugs, but we don't know intent enough to be able to show an inspection highlight. Showing a conversion hint is a great way to show this.

The setting can be changed and overridden by the user, so it can still be disabled.